packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -44,39 +44,40 @@ const FeedEmptyScreen = dynamic(
   () => import(/* webpackChunkName: "feedEmpty" */ './FeedEmptyScreen'),
 );
 
+const FeedFilters = dynamic(() => import('./filters/FeedFilters'));
+
 type FeedQueryProps = {
   query: string;
   queryIfLogged?: string;
   variables?: Record<string, unknown>;
 };
 
-interface FeedOptionalParams {
-  shouldShowMyFeed?: boolean;
+enum MainFeedPage {
+  MyFeed = 'my-feed',
+  popular = 'popular',
+  search = 'search',
+  upvoted = 'upvoted',
+  discussed = 'discussed',
 }
 
-const getPropsByFeed = ({
-  shouldShowMyFeed = false,
-}: FeedOptionalParams = {}): Record<string, FeedQueryProps> => {
-  return {
-    'my-feed': {
-      query: ANONYMOUS_FEED_QUERY,
-      queryIfLogged: FEED_QUERY,
-    },
-    popular: {
-      query: ANONYMOUS_FEED_QUERY,
-      queryIfLogged: shouldShowMyFeed ? ANONYMOUS_FEED_QUERY : FEED_QUERY,
-    },
-    search: {
-      query: ANONYMOUS_FEED_QUERY,
-      queryIfLogged: FEED_QUERY,
-    },
-    upvoted: {
-      query: MOST_UPVOTED_FEED_QUERY,
-    },
-    discussed: {
-      query: MOST_DISCUSSED_FEED_QUERY,
-    },
-  };
+const propsByFeed: Record<MainFeedPage, FeedQueryProps> = {
+  'my-feed': {
```

Comment:
It is strongly-typed using `Record<MainFeedPage, FeedQueryProps>`, so typos should be covered and reported by Typescript in any case. I think this can be a good discussion about whether we use string for the key or use the defined enum for its type. Let me log it down.